lucene/core/src/java/org/apache/lucene/geo/Tessellator.java

```diff
@@ -915,13 +914,14 @@ private static final Node filterPoints(final Node start, Node end) {
       continueIteration = false;
       nextNode = node.next;
       prevNode = node.previous;
-      //We can filter points when they are the same, if not and they are co-linear we can only
-      //remove it if both edges have the same value in .isNextEdgeFromPolygon
-      if (isVertexEquals(node, nextNode)  ||
-          (prevNode.isNextEdgeFromPolygon == node.isNextEdgeFromPolygon &&
+      // we can filter points when:
+      if (isVertexEquals(node, nextNode)  ||   // 1. they are the same,
+         // isVertexEquals(prevNode, nextNode) || // 2.- each one starts and ends in each other
```

Comment:
nit: remove comment

Reply:
ups... that was a left over while testing